lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1903,7 +1903,7 @@ sub _job_stop_cluster {
     return 0 if $job == $self->id;
     my $rset = $self->result_source->resultset;
 
-    $job = $rset->search({id => $job, result => NONE})->first;
+    $job = $rset->find({id => $job, result => NONE}, {rows => 1});
```

Comment:
Adding `rows => 1` won't change that `result` is never added to the `WHERE` clause :)